packages/interactivity/src/router.js

```diff
@@ -78,11 +78,21 @@ const renderRegions = ( page ) => {
 	} );
 };
 
+// Variable to store the current navigation.
+let navigatingTo = '';
+
 // Navigate to a new page.
 export const navigate = async ( href, options = {} ) => {
 	const url = cleanUrl( href );
+	navigatingTo = url;
```

Comment:
What's the logic for using `url` instead of `href` here?

Reply:
That's a good question. 😄 I think I assumed that, as the `url` would be the same for `href` with hashes, both navigations would wait for the same Promise so that they would happen in order and there would be no problem.

I missed considering that:
1. You could use `force` and request the HTML again.
2. Both navigations would trigger a render, which is bad.

I added a failing test in https://github.com/WordPress/gutenberg/pull/54201/commits/83ae05f54bfc36a6ee3f05dab8152cc398542b02 and fixed it in https://github.com/WordPress/gutenberg/pull/54201/commits/578029c7dc17621e214ab47d955db6193152dba2.

Reply:
Yeah, I think it's a bit safer this way. That way, the user will end up with the correct hash in the browser URL (even though we don't support scrolling to the id yet).

Thanks David!